meta/recipes-extended/libnsl/libnsl2_git.bb

```diff
@@ -27,8 +28,12 @@ EXTRA_OECONF += "--libdir=${libdir}/nsl --includedir=${includedir}/nsl"
 do_install_append() {
 	install -d ${D}${libdir}
 	mv ${D}${libdir}/nsl/pkgconfig ${D}${libdir}
+	rm -f ${D}${libdir}/nsl/*.so
+
+	install -d ${D}${sysconfdir}/ld.so.conf.d/
+	install -m 0644 ${WORKDIR}/libnsl2_libs.conf ${D}${sysconfdir}/ld.so.conf.d/
 }
 
-FILES_${PN} += "${libdir}/nsl/*.so.*"
-FILES_${PN}-dev += "${includedir}/nsl ${libdir}/nsl/*.so"
+FILES_${PN} += "${libdir}/nsl/*.so.* ${sysconfdir}/ld.so.conf.d/libnsl2_libs.conf"
+FILES_${PN}-dev += "${includedir}/nsl"
```

Comment:
It is not clear to me why binaries that link against libnsl.so should load the glibc version and not the nsl2 version. Can you explain?

Reply:
If a binary is linking against `libnsl.so`, that means it didn't have dependency on `libnsl2` (or that it didn't use `pkg-config` to find it), in which case it was built/linked with glibc's version of `libnsl.so` - so we want to ensure it will also load with glibc's version.

Reply:
But couldn't there be binaries that do not care what version of the libnsl is on the system (suppose there comes a libnsl.so.3), and for that reason link against libnsl.so?

Reply:
If thats the case, they shouldn't care that they end up using glibc's version.

But if there is a binary that wanted `libnsl.so.1` but linked against `libnsl.so`, it should only load with `glibc`'s `libnsl.so` not `libnsl2`'s.

Reply:
To me, it seems like the inverse. When the distro is at libnsl.so.2, I think binaries can expect that when they link against libnsl.so, they will get the latest nsl. After all, only glibc is not downgraded, not any other binary recipes right?

> But if there is a binary that wanted glibc's libnsl.so and linked against libnsl.so, it should only load with glibc's libnsl.so not libnsl2's.

Before the downgrade, did glibc install a different nsl library? Was it versioned differently?

Reply:
> To me, it seems like the inverse. When the distro is at libnsl.so.2, I think binaries can expect that when they link against libnsl.so, they will get the latest nsl.


Seeing that `libnsl2` bumped abiVersion, there probably were breaking changes in the API, so `libnsl.so.2` may not be  backward compatible with glibc's `libnsl.so`.

My understanding is that a binary should link against the minimum abiVersion that it is compatible with so that future abiVersions (which may have breaking API changes) won't break the binary.

> After all, only glibc is not downgraded, not any other binary recipes right?

Yes, only glibc was downgraded not other recipes.

> Before the downgrade, did glibc install a different nsl library? Was it versioned differently?

glibc-2.33 doesn't build nsl library anymore, it was removed sometime between glibc-2.24 and glibc-2.33.

My concern is that any binaries and/or libraries produced by glibc will have dependency on its own version of `libnsl.so` and if they're loaded with `libnsl2`'s version, they'll break.